perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_def_o365EmailAddresses_mu.java

```diff
@@ -105,36 +103,29 @@ public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attrib
 			throw new WrongReferenceAttributeValueException(attribute, attrUCO, member, null, member, null, "does not contain " + ucoEmail);
 		}
 
-		//check for duplicities among all members and groups (urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu)
-		AttributesManagerBl attributesManagerBl = sess.getPerunBl().getAttributesManagerBl();
-		Set<Pair<Integer, Integer>> memberPairs = attributesManagerBl.getPerunBeanIdsForUniqueAttributeValue(sess, attribute);
-		memberPairs.remove(new Pair<>(member.getId(), 0));
-		if (!memberPairs.isEmpty()) {
-			try {
-				Member memberWithDuplicateEmail = sess.getPerunBl().getMembersManagerBl().getMemberById(sess, memberPairs.iterator().next().getLeft());
-				throw new WrongReferenceAttributeValueException(attribute, attribute, member, null, memberWithDuplicateEmail, null, "some of the email addresses are already assigned.");
-			} catch (MemberNotExistsException e) {
-				throw new ConsistencyErrorException(e);
-			}
-		}
-		Attribute groupO365EmailAddresses = new Attribute(new urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu().getAttributeDefinition());
-		groupO365EmailAddresses.setValue(emails);
-		Set<Pair<Integer, Integer>> groupResourcePairs = attributesManagerBl.getPerunBeanIdsForUniqueAttributeValue(sess, groupO365EmailAddresses);
-		if (!groupResourcePairs.isEmpty()) {
-			try {
-				Pair<Integer, Integer> GroupResourceWithDuplicateEmail = groupResourcePairs.iterator().next();
-				Group groupWithDuplicateEmail = sess.getPerunBl().getGroupsManagerBl().getGroupById(sess, GroupResourceWithDuplicateEmail.getLeft());
-				Resource resourceWithDuplicateEmail = sess.getPerunBl().getResourcesManagerBl().getResourceById(sess, GroupResourceWithDuplicateEmail.getRight());
-				throw new WrongReferenceAttributeValueException(attribute, groupO365EmailAddresses, member, null, groupWithDuplicateEmail, resourceWithDuplicateEmail, "some of the email addresses are already assigned." );
-			} catch (GroupNotExistsException | ResourceNotExistsException e) {
-				throw new ConsistencyErrorException(e);
+		//No need to check duplicities between other members, cause attribute is unique
+		//Check uniqueness between this attribute and groups
+		try {
+			Attribute groupO365EmailAddresses = new Attribute(sess.getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, G_D_O365_EMAIL_ADDRESSES_O365MU_ATTR));
+			groupO365EmailAddresses.setValue(emails);
+			Set<Integer> groupsWithSameAttributeValueSet = sess.getPerunBl().getAttributesManagerBl().getPerunBeanIdsForUniqueAttributeValue(sess, groupO365EmailAddresses).stream().map(Pair::getLeft).collect(Collectors.toSet());
```

Comment:
NIT: When using the stream API, it is a best practice to use the dot notation to separate the input elements and each operation that is performed.
```suggestion
			Set<Integer> groupsWithSameAttributeValueSet = sess.getPerunBl().getAttributesManagerBl().getPerunBeanIdsForUniqueAttributeValue(sess, groupO365EmailAddresses).stream()
					.map(Pair::getLeft)
					.collect(Collectors.toSet());
```

Reply:
Changed.